Add seamless learning docs

phrdang opened the pull request:
- Update top level page order
- Add seamless learning index
- Add seamless learning subpages for relevant projects

pancakereport reviewed (changes requested):
Data Science Curriculum Guide is not a Seamless Learning project so we shouldn't take credit for it. But I definitely think that it should be featured and referenced heavily.

Also, Benji and Celina's work on the section tool (paid for by Seamless Learning) should be featured. Open source repo here: https://github.com/Cal-CS-61A-Staff/sections

phrdang commented:
Thanks for the feedback, I moved the DS curriculum guide into "quick links" under a new section called "Course Software." I also moved the entire Otter section under Course Software.

In order to have arbitrarily nested child pages, I also bumped `just-the-docs` to [v0.10.0](https://just-the-docs.com/changelog/#release-v0100) which also [deprecates the `has_children` front matter field](https://just-the-docs.com/docs/navigation/main/levels/#child-pages), so I removed that everywhere as well.

pancakereport reviewed (approved):
we can always make more changes later, lol
